src/frontend/context/archives-context.js

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useContext, useState, useEffect } from "react";
+import { getArchivedNotesService } from "../services";
+import { useAuth } from "./auth-context";
+
+const ArchivesContext = createContext();
+
+const ArchivesProvider = ({ children }) => {
+    const [archives, setArchives] = useState([]);
+
+    const { auth } = useAuth();
+
+    useEffect(() => {
+        if (auth.status) {
+            (async () => {
+                const response = await getArchivedNotesService(auth.token);
+                if (response !== undefined) {
+                    setArchives(response)
+                }
+            })();
```

Comment:
Just a suggestion, instead of using IIFE , you can declare a function inside useEffect and then call it inside useEffect itself. This will make code more readable. You can refer this, [https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret](https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret)

Reply:
ah thats a good way of doing it thankyou for letting me know